core/src/apps/management/recovery_device/keyboard_bip39.py

```diff
@@ -204,6 +204,7 @@ async def handle_input(self) -> None:
 
             if touch in race.finished:
                 event, x, y = result
+                workflow.idle_timer.touch()
```

Comment:
Maybe this is too philosophical but shouldn't this line happen also for `self.on_timeout`? The rational is that as a user I need to wait for the timeout to enter the letter. Shouldn't that be considered as an activity as well?

But it doesn't really matter in the big picture.

Reply:
Definitely not in `on_timeout`, that would cause more confusion than add value IMHO.
But moving the `touch()` up outside the if is probably a little more correct, for the reason that you mention.
I'd argue that the current behavior is not wrong either: if the user is actively waiting for the button to become available, they _will_ click before auto-locking anyway.

FWIW for #922 I plan to replace this with an `IdleTimer`, and that will probably revert to not touching in `on_timeout`.

Reply:
Yes that is exactly what I meant. I agree. Let's leave it as you have it.